server.js

```diff
@@ -1,38 +1,170 @@
-// import topMusicData from './data/top-music.json'
+import dartPlayers from './data/dartPlayers.json'
+
+dotenv.config()
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const playerSchema = new mongoose.Schema({
+  name: String,
+  country: String,
+  age: Number,
+  dateOfBirth: String,
+  nickname: String,
+  ranking: Number,
+  tourCard: String,
+  careerEarnings: String
+})
+
+const Player = mongoose.model('Player', playerSchema)
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Player.deleteMany();
+
+    await dartPlayers.forEach((item) => {
+      const newPlayer = new Player(item);
+      newPlayer.save();
+    });
+  }
+  seedDatabase();
+}
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
+
+// This function takes an array and a param and checks if that array includes that param. 
+// Used in countries and nickname endpoint. 
+const checkParam = (array, param) => {
+  for (const item of array) {
+    const isIncluded = item.toLowerCase().includes(param.toLowerCase())
+    if (isIncluded) {
+      return true
+    }
+  }
+}
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send(listEndpoints(app))
+})
+
+// Endpoint for all players.
+app.get('/players', async (req, res) => {
+  try {
+    const players = await Player.aggregate(
+      [
+        // Filters for players with a ranking that is less than or equal as the query or 200 as default.
+        { $match: { ranking: { $lte: req.query.ranking ? +req.query.ranking : 200 } } },
+        { $group: { _id: "$name", doc: { $first: "$$ROOT" } } },
+        { $replaceRoot: { newRoot: "$doc" } },
+        { $sort: { ranking: 1 } }
+      ]
+    )
+    res.json(players)
+  } catch {
+    res.status(400).json({ error: 'Something went wrong' })
+  }
+})
+
+// Endpoint for players by id
+app.get('/players/player/:id', async (req, res) => {
+  try {
+    const player = await Player.findById(req.params.id)
+    if (player) {
+      res.json(player)
+    } else {
+      res.status(404).json({ error: 'Player not found' })
+    }
+  } catch {
+    res.status(400).json({ error: 'Invalid player id' })
+  }
+})
+
+// Endpoint that lists all countries. 
+app.get('/countries', async (req, res) => {
+  try {
+    const countries = await Player.distinct('country')
+    res.json(countries)
+  } catch {
+    res.status(400).json({ error: 'Something went wrong' })
+  }
+})
+
+// Endpoint for country by country name
+app.get('/countries/:country', async (req, res) => {
```

Comment:
you are showing the list of players from certain country so your endpoint should be:
`/players/country/:country`